### DISCUSSION TITLE
Some documents not appearing in modern search results - but are in documents folder and are of same type with same metadata

### DISCUSSION THREAD

**Discussion by bobstovall** [microsoft-search/pnp-modern-search#2745] [Q&A] [answered]
**Version used**
4.8.0.0

**Describe the bug**
Thanks for any help, as this is my first time working with SharePoint and pnp modern search, and I'm not a coding expert at all...

Sharepoint modern search worked well at first showing eight PowerPoint files in the results window with two filers off to the side for two categories. Then a day later, with nothing on my end changed, only three PowerPoints showed up in the results window and the rest of the files were not showing up anymore. I checked the document folder, and the files are still there, and meta and everything is the same, but for some reason, the search window is only showing three of the PowerPoint files, not the full eight before filtering. I have re-added new ppt documents to the folder, and they do not show up. I have also re-indexed the library in the search page and rebuilt the site index to no avail.  Any ideas as to what else I can try?

**To Reproduce**
I just deleted all of the PPT files and re-added them to the document library, and the same thing happens. Only three of the files show up out of the eight. I have checked the permissions and everything checks out. 

**Expected behavior**
For modern filtering and search to return all files in the library before filtering.

**Screenshots**
N/A

**Desktop (please complete the following information):**
Chrome and Edge for Windows and Chrome for Mac

**Additional context**
N/A

**Comment by wobba** [comment]
If all files are returned in OOB site search, then it seems like a configuration issue with the web part. Please share your web part setup with data source, query template etc. as that is needed to assist.  If the issue is the same using OOB search, then it's API related.

**Reply by bobstovall** [reply]
Hi, and Thanks for helping!  Everything is showing up in the OOB search...
Here is my Query template:

{searchTerms}
Path:https://armdynamics.sharepoint.com/sites/TheMarketingMediaLibrary/InService Presentations FileType:PPTX

Again, all 8 files are pptx files, but only three of them show up with pnp-modern search filtering.
Thanks again!

**Reply by kasperbolarsen** [reply]
I might have an issue with Presentation part of your query 😉 
Are you trying to filter based on a metadata field or just relying on that term being present in any of the many fields?

**Reply by bobstovall** [reply]
I'm trying to filter the metadata field, specifically the owstaxidmetadataalltagsinfo filter field.

**Reply by kasperbolarsen** [reply]
Which makes me think that there must be a managed metadata field ( most organizations has a DocumentType termset) and therefore the query perhaps should be like owstaxidDocu mentType:Presentation ?

Or even better : owstaxidDocumentType:[theGUIDforthePresentationTerm]. This way your query will not break when somebody decides to rename Presentation to CompanyInfo some day

**Reply by bobstovall** [reply]
Actually, I'm using the Enterprise Keywords field and labeling everything from there... I have a sharepoint site where I'm hosting a few document libraries with photos, pdf's, pptx files, and one with a mix of jpeg, pdf, pptx and word doc files. It was suggested that I use the modern pnp search/filter to better help my organization find what they are looking for. So I used the enterprise keywords field and labeled everything with how they should be filtered. That's my managed metadata field that's being retrieved from owstaxidmetadataalltagsinfo

**Reply by kasperbolarsen** [reply]
That is indeed one way to do it 🙂

In that case a owstaxidmetadataalltagsinfo:Presentation ( or the GUID version) would minimize the risk of false positives 😄

**Comment by bobstovall** [comment, marked as answer]
Thank you!  But still doing that, of the 8 PowerPoint files in that particular document folder, all tagged as "Powerpoint" in the enterprise metadata field, only three of the files are still showing up with five missing in the filtering. It's like 5 of those files won't show up no matter what I do.  Should I create a new document folder and start over fresh with this page?

Once again, thanks so much for your help!

**Reply by kasperbolarsen** [reply]
Ok back basics. Try to query using Path:https://armdynamics.sharepoint.com/sites/TheMarketingMediaLibrary/InService/* FileType:PPTX

And then Path:https://armdynamics.sharepoint.com/sites/TheMarketingMediaLibrary/InService/* owstaxidmetadataalltagsinfo:Presentations

Does this give any hints?

**Reply by bobstovall** [reply]
Yes!  Thanks so much! It's all working perfectly now.
I appreciate you!
